__tests__/diff.test.ts

```diff
@@ -0,0 +1,89 @@
+import { LcovFile } from 'lcov-parse'
+import { analyzeDiffCoverage } from '../src/diff'
+import { ContextInfo, Octokit } from '../src/types'
+import { readFileSync } from 'fs'
+
+describe('diff', () => {
+  it('should correctly calculate coverage percentage for added lines in diff', async () => {
+    const diffString = readFileSync(
+      `${__dirname}/fixture/diff/diff.diff`,
+      'utf-8'
+    )
+    const lcovRaw = readFileSync(
+      `${__dirname}/fixture/diff/parsedLcov.json`,
+      'utf-8'
+    )
+    const lcovFiles: LcovFile[] = JSON.parse(lcovRaw)
+
+    const octokit = {
```

Comment:
https://stackoverflow.com/questions/68605384/jest-mocking-of-octokit-library

jest.mock('@octokit/rest')
const request = () => new Promise((resolve, reject) => {

  resolve({ status: 302, headers: { location: 'mock-url' } });
})
Octokit.mockImplementation(() => ({ request }))

Reply:
We can do this in a follow up thou ...